Record ACTIONX Wells Affected by WELPI Separately

bska opened the pull request:
This way, the simulator layer has more information by which to decide if applying a CTF update is safe/sufficient or if it needs to fully rebuild its internal representation of the simulation wells.

bska commented:
Note: This is the first of a pair of PRs that aims to provide an alternative to OPM/opm-simulators#5624.  The downstream companion PR OPM/opm-simulators#5697 must be reviewed/merged alongside this work.  I am creating the PR in a draft state because it requires a reasonable amount of testing to verify that it actually solves the same problem as OPM/opm-simulators#5642.

bska commented:
jenkins build this opm-simulators=5697 please

akva2 commented:
i can confirm this fixes all the tests that failed in the debug iterator build that was related to the oob indexing. nice!

bska commented:
> i can confirm this fixes all the tests that failed in the debug iterator build that was related to the oob indexing. nice!

Thanks a lot for testing.  I'll fix the unit test that I forgot about in the first version of this and then I'll mark it ready for review.

bska commented:
> > i can confirm this fixes all the tests that failed in the debug iterator build that was related to the oob indexing. nice!
> 
> Thanks a lot for testing. I'll fix the unit test that I forgot about in the first version of this and then I'll mark it ready for review.

I've fixed the unit test so I'm marking this as "ready for review" now.  I've also added the work to the 2024.10 release milestone since the release is slipping and it'd be good to have this in the release as well.

bska commented:
jenkins build this opm-simulators=5697 please
